scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -771,10 +803,54 @@ private void dispatchJobStateUpdated(String owner, NotificationData<JobInfo> not
     }
 
     /**
+     * Dispatch the job state updated event
+     * 
+     * @param owner
+     *            the owner of this job
+     * @param notification
+     *            the data to send to every client
+     */
+    private void dispatchJobUpdatedFullData(JobState job) {
+        try {
+            if (logger.isDebugEnabled()) {
+                jlogger.debug(job.getJobInfo().getJobId(), " event [" + SchedulerEvent.JOB_UPDATED + "]");
+            }
+            for (ListeningUser listeningUserId : identifications.values()) {
+                // if this user has a listener
+                if (listeningUserId.isListening()) {
+                    try {
+                        UserIdentificationImpl userId = listeningUserId.getUser();
+                        // if there is no specified event OR if the specified
+                        // event is allowed
+                        if ((userId.getUserEvents() == null) ||
+                            userId.getUserEvents().contains(SchedulerEvent.JOB_UPDATED)) {
+                            // if this userId have the myEventOnly=false or
+                            // (myEventOnly=true and it is its event)
+                            if (!userId.isMyEventsOnly() ||
+                                (userId.isMyEventsOnly() && userId.getUsername().equals(job.getOwner()))) {
```

Comment:
Yep, but I didn t want to refactor too much this code at this stage. It's a code that is not fully tested and it will require more time to refactor. I prefer to reuse the existing logic now.